web/src/components/AdvancedEditor/hooks/useAutoComplete.ts

```diff
@@ -0,0 +1,104 @@
+import {useCallback} from 'react';
+import {uniqBy} from 'lodash';
+import {CompletionContext} from '@codemirror/autocomplete';
+import {syntaxTree} from '@codemirror/language';
+import {
+  completeIdentifierAfter,
+  completePseudoSelectorAfter,
+  completeValueAfter,
+  operatorList,
+  pseudoSelectorList,
+  Tokens,
+} from 'constants/AdvancedEditor.constants';
+import {useAppStore} from 'redux/hooks';
+import AssertionSelectors from 'selectors/Assertion.selectors';
+import {escapeString} from 'utils/Common';
+
+interface IProps {
+  testId: string;
+  runId: string;
+}
+
+const useAutoComplete = ({testId, runId}: IProps) => {
+  const {getState} = useAppStore();
+
+  const getAttributeList = useCallback(() => {
+    const state = getState();
+    const defaultList = AssertionSelectors.selectAllAttributeList(state, testId, runId);
```

Comment:
I don't think you need the state here. Does this work?
```ts
const defaultList = useAppSelector(state => AssertionSelectors.selectAllAttributeList(state, testId, runId));
```
Maybe the `getAttributeList` is not needed, I mean you can call the `useAppSelector` directly in your return fn, but not a big deal I'm ok with that.

Reply:
Yes, you are right, I had an async process that was causing the callback to update a lot of times which was re-instantiating the extension and creating a weird UX bug. Let me try it again

Reply:
So, what is happening here is that after moving that to an upper selector the callback changes every time the defaultList changes and causes an extensions change which is the reason for this blinking. 

![2022-07-06 1 36 33 p m](https://user-images.githubusercontent.com/11051031/177619680-d755efa7-4f0b-43f5-97bc-d577f090b9c3.gif)

Reply:
But the getState function never changes that's why the blinking doesn't happen when using that approach